### PR TITLE
Color run_simple's terminal output based on HTTP codes

### PR DESCRIPTION
**Pull request by dufferzafar**
This is similar to django's runserver.

Screenshot:

Screenshot

---

It requires termcolor & colorama currently, but consider this just a proof-of-concept, and a work in progress since the code is hacked together. If this functionality is desired, I'll make whatever changes are required to meet the code standards of the project.

I'm not too sure, but I guess #553 refers to this sort of functionality.

### CODE REVIEW

File: werkzeug/serving.py [hunk 1]
```diff
@@ -278,9 +282,42 @@ def log_message(self, format, *args):
...
     def log(self, type, message, *args):
-        _log(type, '%s - - [%s] %s\n' % (self.address_string(),
-                                         self.log_date_time_string(),
-                                         message % args))
+        msg = '%s - - [%s] %s' % (self.address_string(),
+                                  self.log_date_time_string(),
+                                  message % args)
+
+        # HTTP Status Code
+        code = str(args[1])
```
Comment (untitaker) [inline werkzeug/serving.py]: I think this logic should be in `log_request`.

File: werkzeug/serving.py [hunk 2]
```diff
@@ -278,9 +282,42 @@ def log_message(self, format, *args):
...
+            msg = colored(msg, color='white')
+
+        # 304 - Resource Not Modified
+        elif code == '304':
+            msg = colored(msg, color='cyan')
+
+        # 3xx - Redirection
+        elif code[0] == '3':
+            msg = colored(msg, color='green')
+
```
Comment (untitaker) [inline werkzeug/serving.py]: Too many spaces IMO, please move the comments inline to be after the `:` and remove empty lines inbetween.

**Comment by davidism** [comment]
Maybe [click's color support](http://click.pocoo.org/6/utils/#ansi-colors) can be used?

**Comment by dufferzafar** [comment]
@untitaker What about adding two new dependencies? That's what I'm worried about the most. Is that fine?

**Comment by untitaker** [comment]
@davidism Click uses colorama, and Werkzeug doesn't depend on either

@dufferzafar Yeah, it should probably be an optional one. :laughing: So, `try-except` around your imports, and just don't do any coloring if it couldn't be imported. Also you might want to add your modules to `extras_require`.

**Comment by untitaker** [comment]
Also since colorama is just a Windows compat library I think we don't need to import or use it in Werkzeug at all. The user can do that themselves before calling `run_simple`.

**Comment by dufferzafar** [comment]
@untitaker Please have a look now.

File: werkzeug/serving.py [hunk 3]
```diff
@@ -43,6 +43,11 @@
 import signal
 
 try:
+    import termcolor
+except ImportError:
+    pass
```
Comment (untitaker) [inline werkzeug/serving.py]: If termcolor couldn't be imported, then accessing it later in `if termcolor` will crash.

File: setup.py [hunk 4]
```diff
@@ -111,6 +111,7 @@ def run(self):
     packages=['werkzeug', 'werkzeug.debug', 'werkzeug.contrib'],
     extras_require={
         'watchdog': ['watchdog'],
+        'termcolor',
```
Comment (untitaker) [inline setup.py]: That is invalid syntax.

**Comment by untitaker** [comment]
Cool, could you add a changelog entry as well? Thanks!

**Comment by untitaker** [comment]
Thanks!

**Comment by davidism** [comment]
Needs documentation too, something like: "To get colored request logs, install termcolor (with link), and colormama (with link) too if you're on Windows."

**Comment by untitaker** [comment]
Ack, @dufferzafar could you file a new PR for the docs? Thanks!

**Pull request merged**